opendatahub-io/data-science-pipelines
# cherry-pick from kubeflow/pipelines#10550

**Description of your changes:**
Cherry pick from upstream.

fix(Backend + SDK): Add missing optional field to SecretAsVolume and ConfigMapAsVolume.

**Testing instructions:**
* Deploy all components
* Check if all pods are up and running
* As a bonus, run iris-pipeline and see if it finishes successfuly

**Checklist:**
